core/src/foundations/ionic.vars.scss

```diff
@@ -0,0 +1,116 @@
+/**
+ * This is an auto-generated file, please do not change it directly.
+ * Ionic Design System
+ */
+
+$ionic-color-primary-10: #f7faff;
```

Comment:
What's the purpose of these sass variables? On a closer inspection they will be statically set based on the `_design-tokens.json`. If we reference these variables in our components, developers will not be able to customize the value.

I think these either need to be set to the CSS variable, with the default fallback to the static value or we don't generate them at all and reference the CSS variables directly.

Reply:
The purpose was to follow a similar approach as we already have on the md/ios, where only scss variables are used internally and no global :root is defined by default. The would be used internally at component level and to set the values on the global Ionic the defaults, as suggested here: https://github.com/BenOsodrac/ionic-framework/blob/new-ionic-ds/core/src/themes/ionic/ionic.theme.default.scss

But I'm ok with using only the native CSS Variables, its in fact more inline on how I'm used to do things 👍

Reply:
I think it makes sense to have scss variables to use internally, but I think they need to be like `$background-color-step` variables: https://github.com/BenOsodrac/ionic-framework/blob/f3e01bb0057f6c917d582efdf4208829cc6eb14f/core/src/themes/ionic/ionic.theme.default.scss#L106.

With the current state, if I had this:
```
/* button.ionic.scss */
:host {
  color: $ionic-color-primary-10;
}
```

When the stylesheet is compiled the generated output for the developer consuming would be:
```
:host {
  color: #f7faff;
}
```

If a developer tries to customize the color palette in their application:
```
:root {
  --ionic-color-primary-10: orange;
}
```

The button stylesheet will still be `#f7faff`. This isn't desired. 

However if we have:
```
$ionic-color-primary-10: var(--ionic-color-primary-10);
```

Then a developer consuming the components can customize the color used in the component globally.

Reply:
Your PR is merged 👍, lets go 🚀